File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/eksportering/VirksomhetMetadataNæringskode5siffer.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.eksportering;

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.felles.Orgnr;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.felles.Kvartal;

public class VirksomhetMetadataNæringskode5siffer {
    public VirksomhetMetadataNæringskode5siffer(Orgnr orgnr, Kvartal kvartal, NæringOgNæringskode5siffer næringOgNæringskode5siffer) {
        this.orgnr = orgnr;
        this.kvartal = kvartal;
        this.næringOgNæringskode5siffer = næringOgNæringskode5siffer;
    }

    private final Orgnr orgnr;
    private final Kvartal kvartal;
    private final NæringOgNæringskode5siffer næringOgNæringskode5siffer;

    public String getOrgnr() { return orgnr.getVerdi(); }
    public int getÅrstall() { return kvartal.getÅrstall(); }
    public int getKvartal() { return kvartal.getKvartal(); }
    public String getNæring() { return næringOgNæringskode5siffer.getNæring(); }
    public String getNæringskode5siffer() { return næringOgNæringskode5siffer.getNæringskode5Siffer(); }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        VirksomhetMetadataNæringskode5siffer that = (VirksomhetMetadataNæringskode5siffer) o;

        if (!orgnr.equals(that.orgnr)) return false;
        if (!kvartal.equals(that.kvartal)) return false;
        return næringOgNæringskode5siffer.equals(that.næringOgNæringskode5siffer);
    }

    @Override
    public int hashCode() {
        int result = orgnr.hashCode();
        result = 31 * result + kvartal.hashCode();
        result = 31 * result + næringOgNæringskode5siffer.hashCode();
        return result;
    }
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domenemodeller/VirksomhetMetadataNæringskode5siffer.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller

class VirksomhetMetadataNæringskode5siffer(
    private val orgnr: Orgnr,
    private val årstallOgKvartal: ÅrstallOgKvartal,
    private val næringOgNæringskode5siffer: NæringOgNæringskode5siffer
) {
    fun getOrgnr(): String {
        return orgnr.verdi
    }

    val Årstall: Int
        get() = årstallOgKvartal.årstall
    val kvartal: Int
        get() = årstallOgKvartal.kvartal
    val næring: String?
        get() = næringOgNæringskode5siffer.næring
    val næringskode5siffer: String?
        get() = næringOgNæringskode5siffer.næringskode5Siffer

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as VirksomhetMetadataNæringskode5siffer
        if (orgnr != that.orgnr) return false
        return if (årstallOgKvartal != that.årstallOgKvartal) false else næringOgNæringskode5siffer == that.næringOgNæringskode5siffer
    }

    override fun hashCode(): Int {
        var result = orgnr.hashCode()
        result = 31 * result + årstallOgKvartal.hashCode()
        result = 31 * result + næringOgNæringskode5siffer.hashCode()
        return result
    }
}
